core/src/main/scala/neotypes/CypherStringInterpolator.scala

```diff
@@ -17,25 +18,33 @@ object CypherStringInterpolator {
     val params = parameters.iterator
 
     @annotation.tailrec
-    def loop(paramNext: Boolean, output: List[DeferredQueryBuilder.Part]): List[DeferredQueryBuilder.Part] =
+    def loop(paramNext: Boolean, nextSubQuery: Int, output: List[DeferredQueryBuilder.Part]): List[DeferredQueryBuilder.Part] =
       if (paramNext && params.hasNext) {
         params.next() match {
-          case QueryArg.Param(value)      => loop(paramNext = false, output :+ DeferredQueryBuilder.Param(value))
-          case QueryArg.CaseClass(params) => loop(paramNext = false, output ++ caseClassParts(params))
+          case QueryArg.Param(value)          => loop(paramNext = false, nextSubQuery, output :+ DeferredQueryBuilder.Param(value))
+          case QueryArg.CaseClass(params)     => loop(paramNext = false, nextSubQuery, output ++ caseClassParts(params))
+          case QueryArg.QueryBuilder(builder) => loop(paramNext = false, nextSubQuery + 1, output ++ subQueryParts(builder, nextSubQuery))
         }
       } else if (queries.hasNext) {
-        loop(paramNext = true, output :+ DeferredQueryBuilder.Query(queries.next()))
+        loop(paramNext = true, nextSubQuery, output :+ DeferredQueryBuilder.Query(queries.next()))
       } else {
         output
       }
 
-    val queryParts = loop(paramNext = false, Nil)
+    val queryParts = loop(paramNext = false, 1, Nil)
```

Comment:
It would be good to use named parameters for this initial call to `loop` and maybe rename `output` to `acc`